ONSdigital/eq-survey-runner
# Database Session management and RDS Integration

### Changes

Various changes to support RDS integration:
1. Only store a randomly generated session id in the cookie. This eq-session-id is generated using UUID4
2. Create a mapping between this session id and the actual user id and store this in a database table. Then when the request arrives, extract the session id, load the user id from the table and use that to load the questionnaire data. 
3. Add the various postgres dependencies
4. Remove the Flask server side sessions code and dependency (as it doesn't work with postgres)
### How to test
1. Enable server side storage

```
export EQ_SERVER_SIDE_STORAGE=True
```
1. Partially complete a survey, then check you can return to it later. (You can use the dev page to do this)
2. Check you can partially complete different surveys (again use the dev page) and that the data does not get re-used between surveys.
### Who can review

Anyone apart from @warrenbailey
